src/components/light.js

```diff
@@ -79,9 +68,21 @@ module.exports.Component = registerComponent('light', {
       return;
     }
 
-    // No light yet. Create and add light.
-    this.light = getLight(data);
-    if (this.light) {
+    // No light yet or light type has changed. Create and add light.
+    this.setLight(this.data);
+  },
+
+  setLight: function (data) {
```

Comment:
don't think we need to pollute the prototype to add a method that's only used once. and `setLight` name doesn't 100% accurately describe the side-effects